syntax="proto3";

package DataAgent;

// The Data Agent service definition.
service da {

    //**********External Interfaces***************
    // GetConfig external interface
    rpc Config (ConfigReq) returns (ConfigResp) {}

    // GetQuery external interface
    rpc Query (QueryReq) returns (QueryResp) {}

    // GetBlob external interface
    rpc GetBlob (BlobReq) returns (stream Chunk) {}
}

// The request message for GetBlob
message BlobReq {
    string imgHandle = 1;
}

// Chunk message for GetBlob
message Chunk {
    bytes chunk = 1;
}

// The request message for GetQuery
message QueryReq {

}

// The response message for GetQuery
message QueryResp {

}

// The request message for GetConfig
message ConfigReq {

}

// The response message for GetConfig
message ConfigResp {

}
